lib/Consul/API/Agent.pm

```diff
@@ -77,6 +81,10 @@ sub force_leave {
 sub check_register {
     my ($self, $check, %args) = @_;
     croak 'usage: $agent->check_register($check, [%args])' if grep { !defined } ($check);
+    {
+        local $Carp::CarpInternal{ (__PACKAGE__) } = 1;
```

Comment:
Ah, you know, I've never given Carp::Clan a chance.  I will now endeavor to add it to my toolbelt.